text/0000-deprecate-http-tx.md

```diff
@@ -0,0 +1,123 @@
+
+- With v5.0.0 wallets will only accept connections from localhost by default.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Forces changes to existing transaction building workflows that use HTTP(S)
+
+- Remaining transaction options (Slatepack via Tor and armored message) may not be appealing for services already exclusively using HTTP(S)
+
+- Forces "always on" wallet listeners to use Tor or to build out logic to support handling armored slates in the UX flow
+
+- Potentially makes a technology that is already challenging for services to adopt even more challenging to adopt
+
+- Those that do not value privacy as a first-order objective may see this as an unnecessary inconvenience
+
+- Friction could cause some services to avoid Grin completely rather than adding support for a new transaction method
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+By not deprecating HTTP(S) an opportunity is lost to provide an additional layer of privacy to present and future Grin supporters.
+
+Deprecating HTTP only while continuing to support HTTPS would not likely be viable. As currently implemented, removing HTTP support would also mean removing HTTPS support. Allowing HTTPS while blocking HTTP may beyond the scope of `grin-wallet` configuration.
+
+# Prior art
+[prior-art]: #prior-art
+
+A [general comparison of transaction methods for Grin](https://github.com/mimblewimble/grin-pm/issues/283) is available and provides many considerations when weighing possible transaction methods, including HTTP(S).
+
+Bitcoin previously used a similar transaction method (send to IP) which was deprecated as well for many valid reasons. While the arguments presented in this RFC are primarily derived from the objective of delivering privacy, there are other good reasons previously discussed in the context of Bitcoin. These discussions can be found partially in the [bitcointalk post](https://bitcointalk.org/index.php?topic=9334.0) accompanying the [PR](https://github.com/bitcoin/bitcoin/pull/253) that removed this feature for Bitcoin.
+
+Note that Grin is not quite in the same position as Bitcoin was when they deprecated this method as an alternative method existed that did not require interaction was available (P2PKH). Grin will still require interaction for the foreseeable future, either directly via Tor or asynchronously via Slatepack armored messages.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- Is it acceptable to the ecosystem to remove a commonly used transaction method?
+
+- Will the adoption of this RFC cause an unacceptable loss of support from existing adopting services?
+
+- Will deprecating HTTP(S) grow or reduce adoption in the next 6-12 months? 1-3 years? 5-10 years?
+
+- Should sending to non-onion addresses via `grin-wallet send -d` be blocked once HTTP(S) is fully deprecated or should it only prompt a warning as written above?
```

Comment:
> - Should sending to non-onion addresses via `grin-wallet send -d` be blocked once HTTP(S) is fully deprecated or should it only prompt a warning as written above?

As per previous comment, my understanding is that it should be fully blocked. Others?

Reply:
This probably depends on what makes sense and is possible with the implementation of the deprecation, will defer again to @yeastplume for what makes sense based on the existing methods. At the minimum we may just want to block all user-facing accessibility, even if it may still technically be possible for software using grin-wallet.